### PR TITLE
Fix histograms data range change.

### PR DESCRIPTION
**Pull request by ivanmalagon**
Related to https://github.com/CartoDB/cartodb/issues/12618

Related PRs:
- cartodb.js https://github.com/CartoDB/cartodb.js/pull/1719
- deep-insights.js https://github.com/CartoDB/deep-insights.js/pull/577

histograms

When there is a map with several widgets, after applying a filter in one of them, the data returned by the other widgets adapts to the filter. It changes its data range. This is the bug fixed in this PR.

**Acceptance**

- Add several widgets to a map. Histograms and time-series.
- Apply several combinations of filters to those widgets. Assert that the range data doesn't change in all widgets.
- Do it with `show totals` options on and off.
- While in Builder, modify the widget properties: bins, aggregation, column... with `show totals`on. Check that totals (the grey bars behind) reflect those changes, showing the proper visualization.
- Check everything in embed page: no filters, filters applied, changing timezone...
- To sum up, play A LOT with histogram and time-series widgets and check that the data range of all of them doesn't change.

### CODE REVIEW

**Comment by xavijam** [comment]
I feel I've found something weird, but not related to this PR. Time series widget is being rendered each time any filter is applied or map center has changed. Instead of that, only bars should change their height, as we do with the histograms. Any clue @ivanmalagon or @rubenmoya if we have introduced recently or it is happening from the beginning of the new changes?

**Comment by xavijam** [comment]
Found another bug, but again, not related :( -> https://github.com/CartoDB/cartodb/issues/12645

**Comment by ivanmalagon** [comment]
Both bugs mentioned by @xavijam fixed.

- We had a whole render when data changed in time-series that was unnecessary. Now the bars change their height without rendering the whole widget.

- When the bins are changed the histogram widgets gets reset. We weren't doing that and that caused the chart to be bad painted.

**Comment by ivanmalagon** [comment]
retest this please

**Comment by ivanmalagon** [comment]
retest this please
